pinot-core/src/test/java/org/apache/pinot/queries/DistinctQueriesTest.java

```diff
@@ -685,6 +694,18 @@ public void testDistinctWithFilter()
         sortedResults.add(new Record(new Object[]{"California", "San Mateo", 400000}));
         sortedResults.add(new Record(new Object[]{"California", "Sunnyvale", 300000}));
         runQueryInterSegmentWithOrderBy(orderByQuery, sortedResults, new String[]{"State", "City", "SaleAmount"});
+
+        String emptyResultQuery =
```

Comment:
I think this is the scenario when the overall result set of DISTINCT query at table level is empty and was throwing NPE as DistinctTable was null. 

Can we please also add a test where the overall result at table level is non-empty but one or more segment returns empty?

Then we have everything  covered.

Reply:
Now that I think more about it, I don't think this solution will work for the case when result from all segments on a single server is empty but across the servers is non empty. 

The current solution would have sent. a Distinct table from the server with string types. However when the broker reduces distinct tables across server, it might choose the result column types as string if the server that sent empty distinct tables happens to be the 0th server. The broker has to decide the result column types for.the final response and it does that upfront by looking at the column types from first server's table since it assumes they will be the same (they should be in general cases except for this corner case). This will be wrong.

Reply:
We need to introduce some hacky way of knowing this scenario in the broker.reduce code.

Reply:
As long as it is using the `DistinctAggregationFunction.merge()` to merge the DistinctTables, it is handled. Check the fix in  `DistinctAggregationFunction.merge()`.
Let me try to enhance the test to cover that.